docs/internals/frontends.rst

```diff
@@ -438,19 +443,82 @@ The same archive with more information (``borg info --last 1 --json``)::
 File listings
 +++++++++++++
 
-Listing the contents of an archive can produce *a lot* of JSON. Since many JSON implementations
-don't support a streaming mode of operation, which is pretty much required to deal with this amount of
-JSON, output is generated in the `JSON lines <http://jsonlines.org/>`_ format, which is simply
-a number of JSON objects separated by new lines.
-
-Each item (file, directory, ...) is described by one object in the :ref:`borg_list` output.
+Each archive item (file, directory, ...) is described by one object in the :ref:`borg_list` output.
 Refer to the *borg list* documentation for the available keys and their meaning.
 
 Example (excerpt) of ``borg list --json-lines``::
 
     {"type": "d", "mode": "drwxr-xr-x", "user": "user", "group": "user", "uid": 1000, "gid": 1000, "path": "linux", "healthy": true, "source": "", "linktarget": "", "flags": null, "mtime": "2017-02-27T12:27:20.023407", "size": 0}
     {"type": "d", "mode": "drwxr-xr-x", "user": "user", "group": "user", "uid": 1000, "gid": 1000, "path": "linux/baz", "healthy": true, "source": "", "linktarget": "", "flags": null, "mtime": "2017-02-27T12:27:20.585407", "size": 0}
 
+Archive Differencing
+++++++++++++++++++++
+
+Each difference item is described by one object in the :ref:`borg_diff` output.
+The keys in the *borg diff* output are:
+
+path:
+    the filename/path of the file or directory object.
+
+changes:
+    a list of 'changeset' objects describing the changes between the two instances of the object, ``path``, from the
+    two archives.
+
+The keys of a 'changeset' object are:
+
+type:
+  identifies the type of changeset and will contain one of these values
+  
+  - *modified* - file contents changed
+  - *added* - this file was added
+  - *removed* - this file was removed
+  - *added directory* - this directory was added
+  - *removed directory* - this directory was removed
+  - *added link* - this symlink was added
+  - *removed link* - removed symlink
+  - *changed link* - symlink target changed.
+  - *mode* - file/directory/link mode and/or permissions changed. Note - this could indicate a change from a
+    file/directory/link type item to a different type (file/directory/link) item.
+  - *owner* - user and/or group ownership changed.
+
+size:
+    if ``type`` is *added* or *removed*, then the ``size`` key provides the size of the added or removed item.
+
+added:
+    if ``type`` is *modified* and chunk ids can be compared, then ``added`` and ``removed`` keys indicate the amount
+    of data 'added' and 'removed'.
```

Comment:
not sure this description is sufficient to precisely understand what's happening.

maybe check the docs of the non-json diff, is it better and could it be used here also?